### DISCUSSION TITLE
Feature request: `useMutation`

### DISCUSSION THREAD

**Discussion by cawa-93** [wobsoriano/trpc-nuxt#99] [Ideas]
Next to the `useQuery`, a similar `useMutation` is critically needed .🙏🏻
```ts
const {mutate: mutatePoint} = $client.route.point.useMutation()
```

**Comment by ozum** [comment]
Hi @cawa-93, [`useQuery`](https://trpc.io/docs/useQuery) is not from `vue-query`. It is a wrapper around nuxt's [`useAsyncData`](https://nuxt.com/docs/api/composables/use-async-data). So, there is no `useMutation` counterpart available.

However, `tRPC Nuxt` already supports `tRPC` vanilla `mutate`. I also couldn't see it in the docs but found it in the source code.

```ts
const result = await $client.login.mutate({ name: "client" });
```

@wobsoriano, a table like below in the docs may be helpful for new users because I searched the same question as the original poster.

|tRPC Nuxt Method|tRPC method|Leveragesfrom Nuxt|
|---|---|---|
|useQuery()|[Vanilla `query`](https://trpc.io/docs/vanilla)|[useAsyncData](https://nuxt.com/docs/api/composables/use-async-data)|
|query|[Vanilla `query`](https://trpc.io/docs/vanilla)|none|
|mutate|[Vanilla `mutate`](https://trpc.io/docs/vanilla)|none|

@cawa-93, also have a look at #65 which adds partial `vue-query` support, which adds `useQuery` and `useMutation`. (This PR is experimental and alpha quality)
